src/Altinn.Notifications.Email.Core/Status/EmailSendResultMapper.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Azure.Messaging.EventGrid.SystemEvents;
+
+namespace Altinn.Notifications.Email.Core.Status;
+
+/// <summary>
+/// Mapper handling parsing to EmailSendResult
+/// </summary>
+public static class EmailSendResultMapper
+{
+    /// <summary>
+    /// Parse AcsEmailDeliveryReportStatus to EmailSendResult
+    /// </summary>
+    /// <param name="deliveryStatus">Delivery status from Azure Communication Service</param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentException">Throws exception if unknown delivery status</exception>
+    public static EmailSendResult ParseDeliveryStatus(AcsEmailDeliveryReportStatus? deliveryStatus)
+    {
+        if (deliveryStatus == null)
```

Comment:
Would a switch be easier to read here?

Reply:
Alternatively a dictionary and a lookup would be less lines of code.

Reply:
Changed to switch with a little workaround